src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -38,6 +38,12 @@ public partial class HostBuilder : IHostBuilder
         private IServiceProvider? _appServices;
         private PhysicalFileProvider? _defaultProvider;
 
+        [RequiresDynamicCode(Host.RequiresDynamicCodeMessage)]
+        public HostBuilder()
```

Comment:
just wondering, does the attribute cannot be applied to the class fields?

Reply:
No, it can only be applied to methods and the whole class.

https://github.com/dotnet/runtime/blob/e71a9583b4d6c9bd97edd87cda7f98f232f63530/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/RequiresDynamicCodeAttribute.cs#L13-L19

Here I didn't want to mark the whole class, as that has other side-effects with the analyzer - like warning if some other code is using static methods.

I kind of wish just marking the `ctor` as `RequiresDynamicCode` suppressed the code in the field initializer, but I didn't see that happening. @tlakollo - is this supposed to work?

Reply:
It wouldn't a field initializer will not only call the `.ctor` but also the `.cctor`, we don't allow to annotate the `.cctor` directly and the workaround is to annotate the whole class.

Reply:
>  a field initializer will not only call the .ctor but also the .cctor

Yes, it will call the .cctor, but the .cctor doesn't have any "unsafe" code in it. Just the `.ctor` would, with an instance field initializer.

Reply:
Got it. So what you mean is that you would want something like:

```C#
...
private IServiceFactoryAdapter _serviceProviderFactory = new ServiceFactoryAdapter<IServiceCollection>(new DefaultServiceProviderFactory());
...
[RequiresDynamicCode(Host.RequiresDynamicCodeMessage)]
public HostBuilder() {}
...
```
To suppress the warning in the instance field?
If that is the case, yes currently that's not implemented but sounds like a feature that could be done.

Reply:
Correct. That is exactly what I was thinking should work, but didn't.

Reply:
Opened https://github.com/dotnet/linker/issues/2970 to track the implementation, although this will need changes in linker/analyzer/nativeAOT

Reply:
> although this will need changes in linker/analyzer/nativeAOT

I think that's an analyzer-only change. IL based analysis will already see this as part of the constructor method body.

Reply:
(We could possibly do a pragma suppression on the field to shut up analyzer.)